cargo-dist/src/backend/installer/docker.rs

```diff
@@ -0,0 +1,203 @@
+    ///
+    /// We want this to be late-bound and to reference DistManifest
+    /// so linkage data can be incorporated.
+    fn apt_runner_deps(&self, dist: &DistGraph, _manifest: &DistManifest) -> Vec<String> {
+        // FIXME: incorporate linkage info to only include Explicit runtime deps
+        // and to also auto-include runtime deps they forgot to mention!
+        // (soft-blocked on doing a sysdeps refactor)
+        let release = dist.release(self.release_idx);
+        let packages: Vec<String> = release
+            .system_dependencies
+            .apt
+            .clone()
+            .into_iter()
+            .filter(|(_, package)| package.0.wanted_for_target(&self.target))
+            .filter(|(_, package)| package.0.stage_wanted(&DependencyKind::Run))
+            .map(|(name, spec)| {
+                if let Some(version) = spec.0.version {
+                    format!("{name}={version}")
+                } else {
+                    name
+                }
+            })
+            .collect();
+        packages
+    }
+
+    /// Compute the data for the dockerfile
+    ///
+    /// Split off from rendering in case it's useful for testing.
+    fn compute_dockerfile(
+        &self,
+        dist: &DistGraph,
+        manifest: &DistManifest,
+    ) -> DistResult<DockerfileInfo> {
+        let apt_deps = self.apt_runner_deps(dist, manifest);
+
+        let bins = self
+            .bins
+            .iter()
+            .map(|file_name| Bin {
+                source: format!("./{file_name}"),
+                name: file_name.clone(),
+            })
+            .collect();
+        let runner = RunnerImage {
+            image: RUNNER_IMAGE.to_owned(),
+            apt_deps,
+            includes: self.includes.clone(),
+        };
+        Ok(DockerfileInfo { bins, runner })
+    }
+
+    /// Render the dockerfile with the given data
+    fn render_dockerfile(&self, templates: &Templates, info: &DockerfileInfo) -> DistResult<()> {
+        let contents = templates.render_file_to_clean_string(TEMPLATE_INSTALLER_DOCKER, &info)?;
+        let dockerfile_path = self.package_dir.join(DOCKERFILE_NAME);
+        LocalAsset::write_new(&contents, dockerfile_path)?;
+        Ok(())
+    }
+
+    /// Build and export the docker image to a tarball with appropriate metadata,
+    /// assuming `render_dockerfile` has already run.
+    fn build_and_export_image(&self, dist: &DistGraph) -> DistResult<()> {
+        let Some(docker) = &dist.tools.docker else {
+            unreachable!("docker didn't exist, tasks.rs was supposed to catch that!");
+        };
+        // FIXME: in theory we can do "build" and "save" in one shot with
+        //
+        //    docker buildx build --output type=docker,dest=/path/to/output.tar
+        //
+        // which is what the Github Action seems to do, but locally (in WSL) i get:
+        //
+        // > ERROR: Docker exporter feature is currently not supported for docker driver.
```

Comment:
I haven't tested, but I believe if you specify the driver to use in the `buildx` command, you can specify `docker-container` and get something that works in one command.

Reply:
can you elaborate?